### PR TITLE
Added LinkList component

### PR DESCRIPTION
**Pull request by paudyalhari**

### CODE REVIEW

File: packages/react-ui-core/package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
     ]
   },
   "devDependencies": {
+    "@rentpath/react-redux-router": "^3.0.2",
```
Comment (tadjohnston) [inline packages/react-ui-core/package.json]: This shouldn't be here in react-ui-core at all.

File: packages/react-ui-core/src/LinkList/LinkItem.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        {text}
+      </Link>
+    )
+  }
+
+  render() {
+    const { className, tagItem, dataTestId } = this.props
+
+    return (
+      <li className={className} data-test-id={dataTestId} data-tag_item={tagItem}>
```
Comment (tadjohnston) [inline packages/react-ui-core/src/LinkList/LinkItem.js]: We use `data-tid` and not `data-test-id`.

File: packages/react-ui-core/src/LinkList/LinkItem.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    text: PropTypes.string.isRequired,
+    className: PropTypes.string,
+    tagItem: PropTypes.string,
+    useAnchorTag: PropTypes.bool,
+    title: PropTypes.string,
+    dataTestId: PropTypes.string,
+  }
+
+  static defaultProps = {
+    tagItem: 'link',
```
Comment (tadjohnston) [inline packages/react-ui-core/src/LinkList/LinkItem.js]: Please do not pass data tags like as props like this.

File: packages/react-ui-core/src/LinkList/LinkList.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  }
+
+  render() {
+    const { links, theme, className, useAnchorTag } = this.props
+
+    return (
+      <ul className={classnames(className, theme.list)}>
+        { links.map(item => (
+          <LinkItem
+            dataTestId={this.createDataTestId(item)}
```
Comment (tadjohnston) [inline packages/react-ui-core/src/LinkList/LinkList.js]: Let's not pass this in directly like this. These should be static ids anyway.

File: packages/react-ui-core/src/LinkList/LinkList.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+            key={item.url}
+            url={item.url}
+            text={item.text}
+            className={classnames(
+              item.className,
+              theme.item,
```
Comment (tadjohnston) [inline packages/react-ui-core/src/LinkList/LinkList.js]: If you're adding theming, you'll need to wrap this component in `react-themed` `@themed()` and add it the correct way. I can help you if you'd like.

**Comment by paudyalhari** [comment]
Removing this PR since we don't want to have dependency on redux in react-ui.

**Comment by mpelzsherman** [comment]
It is unfortunate that we can't use a `Link` component in `react-core`.

**Comment by tadjohnston** [comment]
@mpelzsherman all you have to do is pass in `nodeType: Link` and it will work as it's supposed to 😄

**Pull request closed without merging**